### PR TITLE
Creating a Shopping Cart using Arrays and Objects using vanilla JavaScript

### LINKED ISSUE #3284: Creating a Shopping Cart using Arrays and Objects using vanilla JavaScript
### Proposed title of article
Creating a Shopping Cart using Arrays and Objects using vanilla JavaScript

### Introduction paragraph (2-3 paragraphs):
There are many ways of building a shopping cart. In this tutorial, I will guide you on how to create a shopping cart using arrays and objects.

The approach that we will use looks as follows. 

We will initially have 2 arrays. One holding the curated list of items to buy, and the other one empty, to hold the items in the cart. We will first display the items in the first array. We will then add an item to the second array and increment the count property in each item, on click. We will decrement the counter and delete the item from the array when the counter reaches zero.

We will use the Javascript event listeners to handle these activities.

Before proceeding, you are required to have some knowledge on the following JavaScript methods:
- Array.push()
- Array.pop()
- EventTarget.addEventListener()

### Key takeaways:
By the end of this tutorial, the reader will be able to:
- Display the shopping list and cart.
- Add items to the cart on click.
- Decrement the item count on click using event listeners.
- Delete items from the cart.

### References: 
N/A

### PR DESCRIPTION
**Pull request by twish-254**
Creating a Shopping Cart Using Arrays and Objects Using Vanilla JavaScript
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
This closes #3284

### CODE REVIEW

File: content/articles/JavaSript-Shopping-Cart/index.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+};
+```
+
+We have created 3 functions.
+
+The *appendNode function* will append a new node to the parent node. It will take the parent node and the new node as parameters.
+
+The *getDiv function* will return the div element with the id of a container.
+
+The *createNode function* will create a new node and return it. It will take a node as a parameter.
```
Comment (geoffrey45) [inline content/articles/JavaSript-Shopping-Cart/index.md:88]: Please convert this into an ordered list.

File: content/articles/JavaSript-Shopping-Cart/index.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+### A little CSS
+
+The displayed lists look very basic at this point. Let's fix that.
+
+In your code editor, open the **index.html** file and paste the code below.
+
+```js
+#wrapper { 
+    display: flex;
```
Comment (geoffrey45) [inline content/articles/JavaSript-Shopping-Cart/index.md:142]: Use CSS syntax highlighting: ```css

File: content/articles/JavaSript-Shopping-Cart/index.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+- Listen for clicks on the items container.
+
+- Call the takeAction function to add the item to the cart.
+
+- Call the displayItems function to display the updated list of items in the cart.
+
+- Listen for clicks in the cart container.
+
+- Call the takeAction function to remove the item from the cart.
```
Comment (geoffrey45) [inline content/articles/JavaSript-Shopping-Cart/index.md:267]: Remove the empty lines between these items.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@twish-254 Thank you for submitting your article for review.

Please check the comments below.

**Comment by twish-254** [comment]
@geoffrey45  I have made all the changes requested

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@twish-254 Please check the comments below

**Comment by twish-254** [comment]
@geoffrey45 I have made the changes

File: content/articles/JavaSript-Shopping-Cart/index.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+The `items_array` is an array of objects which will be displayed for selection. A `name` and `id` property is given to each object.
+The `items_array` also has a `count` property, which is the number of items which will be used in the shopping cart.
+
+The cart is an empty array, which will be used to store the items which will be selected from the shopping cart.
+
+### Creating the base functions
+Let's make a few functions to make our lives a little bit easier. 
+
+In your code editor, open the `script.js` file and add the code below.
+
```
Comment (geoffrey45) [inline content/articles/JavaSript-Shopping-Cart/index.md:66]: @twish-254 This statement is redundant throughout your article. Please replace all the appearances after the first one with something similar to this: `Add the following code to script.js`

File: content/articles/JavaSript-Shopping-Cart/index.md [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+Create a new file `index.html` in the working directory. In your code editor, open the `index.html` file and paste the code below.
+
+```html
+<meta charset="UTF-8">
+<meta http-equiv="X-UA-Compatible" content="IE=edge">
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
+<title>Document</title>
+
+<div id="wrapper">
+   <div id="container">
```
Comment (geoffrey45) [inline content/articles/JavaSript-Shopping-Cart/index.md:28]: @twish-254 This Html block is missing the head and body tags. Please add them.

File: content/articles/JavaSript-Shopping-Cart/index.md [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+    items_container.innerHTML = '';
+
+    for (let i = 0; i &lt; items.length; i++) {
+        let item = items[i];
+
+        let item_node = createNode("li");
+        item_node.setAttribute("id", item.id);
+
+        if (item.count &gt; 0) {
+            item_node.innerHTML = `${item.name}
```
Comment (geoffrey45) [inline content/articles/JavaSript-Shopping-Cart/index.md:106]:
```suggestion
        item_node.setAttribute("id", item.id);

        if (item.count > 0) {
            item_node.innerHTML = `${item.name} 
```

File: content/articles/JavaSript-Shopping-Cart/index.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+In your code editor, open the `script.js` file and add the code below.
+
+```javascript
+function displayItems(items, container) {
+    let items_container = getDiv(container);
+    items_container.innerHTML = '';
+
+    for (let i = 0; i &lt; items.length; i++) {
+        let item = items[i];
```
Comment (geoffrey45) [inline content/articles/JavaSript-Shopping-Cart/index.md:100]:
```suggestion
    items_container.innerHTML = '';

    for (let i = 0; i < items.length; i++) {
        let item = items[i];
```

File: content/articles/JavaSript-Shopping-Cart/index.md [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+    // code in the previous block is still there
+
+    function takeAction(container) {
+        container.addEventListener("click", function (event) {
+            let item_id = event.target.id;
+
+            if (item_id !== "items" &amp;&amp; item_id !== "badge") {
+                let item = items_array.filter(function (item) {
+                    return item.id == item_id;
+                })[0];
```
Comment (geoffrey45) [inline content/articles/JavaSript-Shopping-Cart/index.md:222]:
```suggestion
            if (item_id !== "items" && item_id !== "badge") {
                let item = items_array.filter(function (item) {
                    return item.id == item_id;
                })[0];
```

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@twish-254 There are some Html characters in your JavaScript code. Please check the suggestions below.

**Comment by ahmadmardeni1** [comment]
Great article @twish-254 
Thank you for the review @geoffrey45

**Pull request merged**